### DISCUSSION TITLE
dotnet local tool support for NuGet

### DISCUSSION THREAD

**Discussion by xt0rted** [community/community#13825] [Code Security]
NuGet updates don't include [dotnet local tools](https://docs.microsoft.com/en-us/dotnet/core/tools/global-tools#install-a-local-tool) which are saved in `.config/dotnet-tools.json` (I believe this file can be placed anywhere, not just in the `.config` folder).

For now I'm working around this with a workflow that reads all the tools from that file and dispatches their info to another workflow that checks for updates and creates a PR if an update was found. It'd be great to not need this workaround though.

A demo of the workflows can be found in [xt0rted/dotnet-tool-update-test](https://github.com/xt0rted/dotnet-tool-update-test).

**Comment by xenosie** [comment]
Create a lightweight CLI tool (e.g., ToolUpdater) to automate checks:

-----C#---------------
var manifest = JsonSerializer.Deserialize<Manifest>(File.ReadAllText("dotnet-tools.json"));
foreach (var tool in manifest.Tools)
{
    var latestVersion = await NuGetClient.GetLatestVersion(tool.PackageId);
    if (latestVersion > tool.Version) 
        File.WriteAllText("PR_TRIGGER.txt", $"Update {tool.PackageId}");
}
---------------------------
By combining native CLI commands, optimized GitHub Actions, and strategic manifest management, you can eliminate custom workflow complexity while maintaining robust update automation. For reference implementations, your [demo repository](https://github.com/xt0rted/dotnet-tool-update-test) could integrate these patterns directly.
